Run unit tests for Prefect

### Overview
We need to be able to run unit tests for Prefect. Currently, `docker-compose run prefect pytest` is failing.

We can follow the unit test infra setup that the `api` is using. Prefect does seem to have a few unit tests, but we cannot use docker to run them right now.

### Action Items
- [x] Add `pytest` to Prefect's requirements.txt file. `pytest` is already in the Pipfile, but we are using requirements.txt in the Dockerfile.
- [x] See if `docker-compose run prefect pytest` works after that

added `pytest==7.1.2` to the end of /server/prefect/requirements.txt
ran `docker compose build` to rebuild
then `docker compose run prefect pytest` works and returns the following:

[+] Running 2/2

 ⠿ Network 311_data_default  Created                                                            0.1s

 ⠿ Container 311-postgres    Created                                                            0.2s

[+] Running 1/1

 ⠿ Container 311-postgres  Started                                                              0.4s

======================================== test session starts ========================================

platform linux -- Python 3.7.13, pytest-7.1.2, pluggy-1.0.0

rootdir: /app, configfile: setup.cfg, testpaths: tests

collected 6 items                                                                                   



tests/test_tasks.py F.....                                                                    [100%]



============================================= FAILURES ==============================================

_________________________________________ test_last_updated _________________________________________



    def test_last_updated():

>       runner = TaskRunner(task=postgres.get_last_updated)

E       AttributeError: module 'tasks.postgres' has no attribute 'get_last_updated'



tests/test_tasks.py:14: AttributeError

========================================= warnings summary ==========================================

../usr/local/lib/python3.7/site-packages/marshmallow/fields.py:198: 11 warnings

  /usr/local/lib/python3.7/site-packages/marshmallow/fields.py:198: RemovedInMarshmallow4Warning: Passing field metadata as a keyword arg is deprecated. Use the explicit `metadata=...` argument instead.

    RemovedInMarshmallow4Warning,



-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

====================================== short test summary info ======================================

FAILED tests/test_tasks.py::test_last_updated - AttributeError: module 'tasks.postgres' has no att...

============================= 1 failed, 5 passed, 11 warnings in 54.96s =============================

--- a/.github/workflows/build_test_backend.yml
+++ b/.github/workflows/build_test_backend.yml
@@ -30,5 +30,6 @@ jobs:
         run: docker-compose run -e TESTING=True api alembic upgrade head
       - name: Run unit tests
         run: docker-compose run api pytest
+        run: docker-compose run prefect pytest
       # - name: Run Postman tests
       #   run: chmod +x postman/test.sh && postman/test.sh

--- a/server/prefect/requirements.txt
+++ b/server/prefect/requirements.txt
@@ -44,3 +44,4 @@ websocket-client==0.58.0
 websockets==8.1
 zict==2.0.0
 zipp==3.4.1
+pytest==7.1.2

--- a/server/prefect/requirements.txt
+++ b/server/prefect/requirements.txt
@@ -22,6 +22,7 @@ psutil==5.8.0
 psycopg2-binary==2.8.6
 pyee==8.1.0
 pyppeteer==0.2.5
+pytest==7.1.2
 python-box==5.3.0
 python-dateutil==2.8.1
 python-slugify==4.0.1
@@ -44,4 +45,3 @@ websocket-client==0.58.0
 websockets==8.1
 zict==2.0.0
 zipp==3.4.1
-pytest==7.1.2

--- a/server/prefect/tests/test_tasks.py
+++ b/server/prefect/tests/test_tasks.py
@@ -10,13 +10,6 @@ from pendulum import parse
 from tasks import postgres, socrata
 
 
-def test_last_updated():
-    runner = TaskRunner(task=postgres.get_last_updated)
-    state = runner.run()
-    assert state.is_successful()
-    assert state.result > parse('2020-01-01', tz=None)
-
-
 def test_download_data():
     with prefect.context(today=datetime.today().strftime('%Y-%m-%d')):
         new_state = TaskRunner(

